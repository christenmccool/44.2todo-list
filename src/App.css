.App {
  text-align: center;
}

.App-heading {
  font-size: 60px;
}

.App-button {
  margin-top: 80px;
  padding: 4px 50px;
  margin-left: 20px;
  font-size: 20px;
  border-radius: 8px;
  border: 2px darkgrey solid;
  background-color: white;
}
