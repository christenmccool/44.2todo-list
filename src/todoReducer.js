const INITIAL_STATE = {todos:[]};
// const INITIAL_STATE = {todos:[{id: 1, task: "make bed", complete: true}, {id: 2, task: "fold laundry", complete: false}]};


const todoReducer = (state=INITIAL_STATE, action) => {
  switch (action.type) {
    case "ADD_TODO": return {...state, todos: [...state.todos, {...action.payload}]};
    case "REMOVE_TODO": return {...state, todos: state.todos.filter(todo => todo.id !== action.id)};
    case "TOGGLE_COMPLETE": {
      return {...state, todos: state.todos.map(todo => {
        if (todo.id === action.id) return {...todo, complete: !todo.complete};
        return todo;
        })
      }
    }
    case "EDIT_TODO": {
      return {...state, todos: state.todos.map(todo => {
        if (todo.id === action.payload.id) return {...action.payload};
        return todo;
        })
      }
    }
    default: return state;
  }
}

export default todoReducer;