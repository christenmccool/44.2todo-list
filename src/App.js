import React, {useState}  from 'react';
import {useSelector, useDispatch} from 'react-redux';
import { v4 as uuid } from 'uuid';
import ToDoList from './ToDoList';
import ToDoForm from './ToDoForm';
import './App.css';

function App() {
  const [showForm, setShowForm] = useState(false);

  const todos = useSelector(store => store.todos);
  const dispatch = useDispatch();

  const addToDo = (todo) => {
    const newTodo = {...todo, complete: false, id: uuid()};
    dispatch({type: 'ADD_TODO', payload:newTodo});
  }

  const toggleComplete = (id) => {
    dispatch({type: 'TOGGLE_COMPLETE', id});
  }

  const editToDo = (editedTodo) => {
    dispatch({type: 'EDIT_TODO', payload: editedTodo});
  }

  const removeToDo = (id) => {
    dispatch({type: 'REMOVE_TODO', id});
  }

  const handleClick = () => {
    setShowForm(!showForm);
  }


  return (
    <div className="App">
      <h1 className="App-heading">To Do List</h1>
      <ToDoList todos={todos} toggleComplete={toggleComplete} editToDo={editToDo} removeToDo={removeToDo} />
      <button className="App-button" onClick={handleClick}>New To Do Form</button>
      {showForm ? <ToDoForm addToDo={addToDo} /> : null}
    </div>
  );
}

export default App;
