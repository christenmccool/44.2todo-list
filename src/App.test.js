import { render, asFragment,fireEvent } from '@testing-library/react';
import todoReducer from './todoReducer';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import App from './App';


const store = createStore(todoReducer);

test('renders without crashing', () => {
  render(<Provider store={store}><App /></Provider>);
});

test('matches snapshot', () => {
  const {asFragment} = render(<Provider store={store}><App /></Provider>);
  expect(asFragment()).toMatchSnapshot();
});

test('shows heading and button', () => {
  const {queryByText} = render(<Provider store={store}><App /></Provider>);
  expect(queryByText('To Do List')).toBeInTheDocument();
  expect(queryByText('New To Do Form')).toBeInTheDocument();
});

test('show and hide form on button click', () => {
  const {queryByText} = render(<Provider store={store}><App /></Provider>);
  expect(queryByText('Task:')).not.toBeInTheDocument();

  const showBtn = queryByText('New To Do Form');
  fireEvent.click(showBtn);
  expect(queryByText('Task:')).toBeInTheDocument();
  expect(queryByText('Select priority')).toBeInTheDocument();

  fireEvent.click(showBtn);
  expect(queryByText('Task:')).not.toBeInTheDocument();
  expect(queryByText('Select priority')).not.toBeInTheDocument();
});
