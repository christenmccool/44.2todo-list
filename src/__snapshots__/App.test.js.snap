// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1
      class="App-heading"
    >
      To Do List
    </h1>
    <div
      class="ToDoList"
    />
    <button
      class="App-button"
    >
      New To Do Form
    </button>
  </div>
</DocumentFragment>
`;
